tests/model_registry/rest_api/conftest.py

```diff
@@ -47,3 +71,132 @@ def updated_model_artifact(
+
+    Ensures the odh-trusted-ca-bundle ConfigMap is mounted in both MySQL and Model Registry deployments,
+    and the --ssl-ca argument is set to the correct path.
+    """
+    mysql_template = get_model_registry_deployment_template_dict(
+        secret_name=model_registry_db_secret.name,
+        resource_name=DB_RESOURCES_NAME,
+    )
+
+    mysql_template["spec"]["containers"][0]["args"].append(f"--ssl-ca={CA_FILE_PATH}")
+    mysql_template["spec"]["containers"][0]["volumeMounts"].append({
+        "mountPath": CA_MOUNT_PATH,
+        "name": CA_CONFIGMAP_NAME,
+        "readOnly": True,
+    })
+    mysql_template["spec"]["volumes"].append({"name": CA_CONFIGMAP_NAME, "configMap": {"name": CA_CONFIGMAP_NAME}})
+
+    patch = {"spec": {"template": mysql_template["spec"]}}
+
+    with ResourceEditor(patches={model_registry_db_deployment: patch}):
+        with create_model_registry_instance(
+            namespace=model_registry_namespace,
+            name=SECURE_MR_NAME,
+            labels=MODEL_REGISTRY_STANDARD_LABELS,
+            grpc={},
+            rest={},
+            istio=ISTIO_CONFIG_DICT,
+            mysql=model_registry_mysql_config,
+            wait_for_resource=True,
+        ) as mr:
+            mr.wait_for_condition(condition="Available", status="True")
+            yield mr
+
+
+@pytest.fixture
+def local_ca_bundle(request: pytest.FixtureRequest, admin_client: DynamicClient) -> Generator[str, Any, Any]:
+    """
+    Creates a local CA bundle file by fetching the CA bundle from a ConfigMap and appending the router CA from a Secret.
+    """
+    namespace = getattr(request, "param", {}).get("namespace", "test-model-registry-namespace")
+    ca_bundle_path = getattr(request, "param", {}).get("ca_bundle_path", "ca-bundle.crt")
+    cert_name = getattr(request, "param", {}).get("cert_name", "ca-bundle.crt")
+
+    cm = ConfigMap(client=admin_client, name="odh-trusted-ca-bundle", namespace=namespace)
+    ca_bundle_content = cm.instance.data.get(cert_name)
+    with open(ca_bundle_path, "w", encoding="utf-8") as f:
+        f.write(ca_bundle_content)
+
+    try:
+        router_secret = Secret(client=admin_client, name="router-ca", namespace="openshift-ingress-operator")
+        router_ca_b64 = router_secret.instance.data.get("tls.crt")
+        if router_ca_b64:
+            router_ca_content = base64.b64decode(router_ca_b64).decode("utf-8")
+            with open(ca_bundle_path, "r", encoding="utf-8") as bundle:
+                bundle_content = bundle.read()
+            if router_ca_content not in bundle_content:
+                with open(ca_bundle_path, "a", encoding="utf-8") as bundle_append:
+                    bundle_append.write("\n" + router_ca_content)
+    except Exception as e:
+        pytest.fail(f"router-ca secret not found in openshift-ingress-operator. Error: {e}")
+    yield ca_bundle_path
+
+    try:
+        os.remove(ca_bundle_path)
+    except FileNotFoundError:
+        pass
```

Comment:
We should not hit this path, right? If we have problem removing something we created, that should not be hidden.